src/Stratis.Bitcoin.Tests/Utilities/AsyncManualResetEventTest.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Stratis.Bitcoin.Utilities;
+using Xunit;
+
+namespace Stratis.Bitcoin.Tests.Utilities
+{
+    public class AsyncManualResetEventTest
+    {
+        [Fact]
+        public async void AsyncManualResetEvent_WaitAsync()
```

Comment:
what is the purpose of this test? (there is no comment ...)
it seems to test that
- using AMRE does not crash
- WaitAsync eventually completes

however, it does not check
- whether WaitAsync completes when the event is set

also please use lambdas over delegates to be consistent with the rest of the codebase

Reply:
Fixed.